cocos/2d/assets/sprite-frame.ts

```diff
@@ -487,6 +510,35 @@ export class SpriteFrame extends Asset {
         return this._original;
     }
 
+    /**
+     * @en Number of pixels corresponding to unit size in world space (pixels per meter)
+     * @zh 世界空间中的单位大小对应的像素数量（像素每米）
+     */
+    get pixelsToUnit () {
+        return this._pixelsToUnit;
+    }
+
+    /**
+     * @en Local origin position when generating the mesh
+     * @zh 生成 mesh 时本地坐标原点位置
+     */
+    get pivot () {
+        return this._pivot;
+    }
+
+    /**
+     * @en mesh information
+     * @zh mesh 信息
+     */
+    get mesh () {
+        if (!this._mesh) {
+            // If SpriteFrame from load, we need init vertices when use mesh
+            this._initVertices();
+            this.createMesh();
```

Comment:
This is an implicit memory heavy process, please use explicit creation API for mesh, for example make `createMesh` public (do some inside check to avoid repeat mesh creation), and when outer object wants to access mesh, it need to manually make sure mesh is created.